### PR TITLE
Upgrade to Phinx 0.6.5

### PR DESCRIPTION
**Pull request by HavokInspiration**
I upgraded with the idea in mind of having our test suite passing.

There is no guarantee that all new Phinx features between 0.5.3 (the current version pinned) and 0.6.5 (the latest) will work out of the box but I'm thinking we can go from there (with the plugin working as expected) and patch things up as we go.

### CODE REVIEW

File: src/CakeAdapter.php [hunk 1]
```diff
@@ -685,4 +686,49 @@ public function dropDatabase($name)
...
+    {
+        return $this->adapter->getInput();
+    }
+
+    /**
+     * Toggle a migration breakpoint.
+     *
+     * @param MigrationInterface $migration
+     *
+     * @return AdapterInterface
```
Comment (dereuromark) [inline src/CakeAdapter.php]: Shouldnt this then be $this ?

Comment (HavokInspiration) [inline src/CakeAdapter.php]: Yes probably. It's actually the stub generated by PHPStorm. I'll fix it.

File: src/Migrations.php [hunk 2]
```diff
@@ -287,7 +294,8 @@ public function getManager($config = null)
                 );
             }
 
-            $this->manager = new CakeManager($config, $this->output);
+            $input = empty($this->input) ? $this->stubInput : $this->input;
```
Comment (dereuromark) [inline src/Migrations.php]: Shouldnt this be

    $input = $this->input) ?: $this->stubInput;

?

File: src/Migrations.php [hunk 3]
```diff
@@ -235,8 +236,14 @@ public function seed($options = [])
     {
         $this->setCommand('seed');
         $input = $this->getInput('Seed', [], $options);
-        $params = ['default', $input->getOption('seed')];
-        $this->run('Seed', $params, $input);
+
+        $seed = $input->getOption('seed');
+        if (empty($seed)) {
```
Comment (dereuromark) [inline src/Migrations.php]: Ideally, we dont use cloaking where not necessary, so !$seed would be cleaner here.

File: src/Migrations.php [hunk 4]
```diff
@@ -294,7 +302,7 @@ public function getManager($config = null)
                 );
             }
 
-            $input = empty($this->input) ? $this->stubInput : $this->input;
+            $input = $input = $this->input ?: $this->stubInput;
```
Comment (dereuromark) [inline src/Migrations.php]: double $input? :)

File: src/Migrations.php [hunk 5]
```diff
@@ -58,6 +58,14 @@ class Migrations
     protected $command;
 
     /**
+     * Stub input to feed the manager class since we might not have an input ready when we get the Manager using
+     * the `getManager()` method
+     *
+     * @var ArrayInput
```
Comment (dereuromark) [inline src/Migrations.php]: For phpdoc its probably best to have all FQCN and with leading \

File: src/CakeManager.php [hunk 6]
```diff
@@ -172,7 +172,7 @@ public function rollbackToDateTime($environment, \DateTime $dateTime)
         $versionToRollback = $versions[$index];
 
         $this->getOutput()->writeln('Rolling back to version ' . $versionToRollback);
-        $this->rollback($environment, $versionToRollback);
+        $this->rollback($environment, $versionToRollback, $force = false);
```
Comment (HavokInspiration) [inline src/CakeManager.php]: This is wrong. I'll fix it.

**Comment by lorenzo** [comment]
Great job @HavokInspiration !

**Pull request merged**